ingestors/kafka/src/main/scala/hydra/kafka/util/ConsumerGroupsOffsetConsumer.scala

```diff
@@ -0,0 +1,223 @@
+    Serializer.delegate[F, A] {
+      val serializer = {
+        val se = new KafkaAvroSerializer(schemaRegistryClient)
+        se.configure(Map(AbstractKafkaAvroSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG -> "").asJava, isKey)
+        se
+      }
+      (topic: String, data: A) => serializer.serialize(topic, data)
+    }.suspend
+
+  private def processRecord[F[_]: ConcurrentEffect: Logger](
+                                                             cr: CommittableConsumerRecord[F, Option[BaseKey], Option[OffsetAndMetadata]],
+                                                             s: SchemaRegistryClient,
+                                                             destinationTopic: String,
+                                                             dvsInternalKafkaOffsetTopic: String
+                                                           ): fs2.Stream[F, ProducerRecords[Array[Byte], Array[Byte], Unit]] = {
+    ((cr.record.key, cr.record.value) match {
+      case (Some(OffsetKey(_, k)), offsetMaybe) =>
+        val topicMaybe: Option[String] = Option(k.topicPartition.topic())
+        val consumerGroupMaybe: Option[String] = Option(k.group)
+        val consumerKeyMaybe: Option[TopicConsumerKey] = consumerGroupMaybe.flatMap(cg => topicMaybe.map(t => TopicConsumerKey(t, cg)))
+        val consumerValue = offsetMaybe.map(o => Instant.ofEpochMilli(o.commitTimestamp)).map(TopicConsumerValue.apply)
+
+        consumerKeyMaybe match {
+          case Some(consumerKey) =>
+            fs2.Stream.eval(for {
+              topicConsumer <- TopicConsumer.encode[F](consumerKey, consumerValue)
+              topicConsumerOffset <- TopicConsumerOffset.encode[F](
+                TopicConsumerOffsetKey(cr.offset.topicPartition.topic(),cr.offset.topicPartition.partition()),
+                TopicConsumerOffsetValue(cr.offset.offsetAndMetadata.offset())
+              )
+              (key, value) = topicConsumer
+              (offsetKey, offsetValue) = topicConsumerOffset
+              k <- getSerializer[F, GenericRecord](s)(isKey = true).serialize(destinationTopic, Headers.empty, key)
+              v <- getSerializer[F, GenericRecord](s)(isKey = false).serialize(destinationTopic, Headers.empty, value.orNull)
+              offsetK <- getSerializer[F, GenericRecord](s)(isKey = true).serialize(dvsInternalKafkaOffsetTopic, Headers.empty, offsetKey)
+              offsetV <- getSerializer[F, GenericRecord](s)(isKey = false).serialize(dvsInternalKafkaOffsetTopic, Headers.empty, offsetValue)
+            } yield  {
+              val p = ProducerRecord(destinationTopic, k, v)
+              val p2 = ProducerRecord(dvsInternalKafkaOffsetTopic, offsetK, offsetV)
+//              val partition = cr.offset.topicPartition.partition()
+//              myMap = myMap + (partition -> cr.offset.offsetAndMetadata.offset())
+//              val string = myMap.toList.sortBy(_._1).map(m => s"${m._1}:${m._2}").mkString(" | ")
+//              println(string)
```

Comment:
Do you need to delete these lines? Or uncomment them?

Reply:
Whoops, deleting now.